thirdeye/thirdeye-frontend/app/pods/rootcause-aggregates-cache/service.js

```diff
@@ -56,12 +56,43 @@ export default Service.extend({
       return;
     }
 
-    // metrics
+    let metricUrnToOffestAndUrn = {};
     missing.forEach(urn => {
-      return this._fetchSlice(urn, requestContext);
+      const metricUrn = toMetricUrn(urn);
+      if (!(metricUrn in metricUrnToOffestAndUrn)) {
```

Comment:
seems like we could just do:
`const arrayName = metricUrnToOffestAndUrn[metricUrn] || []`

Reply:
Thank you! This is cool.